tools/wpt/wpt.py

```diff
@@ -0,0 +1,115 @@
+import argparse
+import os
+import json
+import sys
+import localpaths
+from browserutils import virtualenv
+
+here = os.path.dirname(__file__)
+wpt_root = os.path.abspath(os.path.join(here, "..", ".."))
+
+
+def load_commands():
+    rv = {}
+    with open(os.path.join(here, "paths"), "r") as f:
+        paths = [item.strip().replace("/", os.path.sep) for item in f if item.strip()]
+    for path in paths:
+        abs_path = os.path.join(wpt_root, path, "commands.json")
+        base_dir = os.path.dirname(abs_path)
+        with open(abs_path, "r") as f:
+            data = json.load(f)
+            for command, props in data.iteritems():
+                assert "path" in props
+                assert "script" in props
+                rv[command] = {
+                    "path": os.path.join(base_dir, props["path"]),
+                    "script": props["script"],
+                    "parser": props.get("parser"),
+                    "help": props.get("help"),
+                    "virtualenv": props.get("virtualenv", True),
+                    "install": props.get("install", []),
+                    "requirements": [os.path.join(base_dir, item)
+                                     for item in props.get("requirements", [])]
+                }
+    return rv
+
+
+def parse_args(commands):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--venv", action="store", help="Path to an existing virtualenv to use")
+    parser.add_argument("--debug", action="store_true", help="Run the debugger in case of an exception")
+    subparsers = parser.add_subparsers(dest="command")
+    for command, props in commands.iteritems():
```

Comment:
Shrug. I guess I ought to make this work with Python 3, but wptrunner doesn't, for example, so it won't be *super* useful.